src/main.rs

```diff
@@ -20,20 +21,28 @@ fn main() -> anyhow::Result<()> {
         _ => "trace",
     })).init();
 
-    let remote_name = args.remote + "/";
-    let remote_name = remote_name.as_str();
     let repo_path = args.repo.as_deref().unwrap_or(".");
     let repo = open_repo(repo_path)?;
 
+    // Obtain remote name from args or use default
+    let remote_name = if let Some(name) = args.remote { name } else {
+        let remotes = repo.remotes()?;
+        if remotes.len() != 1 {
+            return Err(anyhow!(AppError::RemoteNotSpecified));
+        }
+        remotes.get(0).unwrap().to_string()
```

Comment:
is 0 the original branch?

Reply:
i have checked remotes' length, so it's definitly the only remote.